homeassistant/components/klyqa/light.py

```diff
@@ -0,0 +1,781 @@
+"""Support for klyqa lights."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable, Coroutine
+from functools import partial
+import json
+import traceback
+from typing import Any
+
+from klyqa_ctl import klyqa_ctl as api
+
+from homeassistant.components.group.light import LightGroup
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_BRIGHTNESS_PCT,
+    ATTR_COLOR_TEMP,
+    ATTR_EFFECT,
+    ATTR_HS_COLOR,
+    ATTR_RGB_COLOR,
+    ATTR_RGBWW_COLOR,
+    ATTR_TRANSITION,
+    ENTITY_ID_FORMAT,
+    ColorMode,
+    LightEntity,
+    LightEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.helpers import (
+    area_registry as ar,
+    device_registry as dr,
+    entity_registry as er,
+)
+from homeassistant.helpers.area_registry import SAVE_DELAY, AreaEntry
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.entity_registry import EntityRegistry, RegistryEntry
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+import homeassistant.util.color as color_util
+from homeassistant.util.color import (
+    color_temperature_kelvin_to_mired,
+    color_temperature_mired_to_kelvin,
+)
+
+from .const import (
+    CONF_POLLING,
+    CONF_SYNC_ROOMS,
+    DOMAIN,
+    EVENT_KLYQA_NEW_LIGHT,
+    EVENT_KLYQA_NEW_LIGHT_GROUP,
+    LOGGER,
+)
+from .datacoordinator import HAKlyqaAccount, KlyqaDataCoordinator
+
+NoneType = type(None)
+
+
+TIMEOUT_SEND = 11
+
+SUPPORT_KLYQA = LightEntityFeature.TRANSITION
+
+
+async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
+    """Expose light control via state machine and services."""
+    return True
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Async_setup_entry."""
+    klyqa: HAKlyqaAccount | None = None
+
+    if entry.entry_id not in hass.data[DOMAIN].entries:
+        hass.data[DOMAIN].entries[entry.entry_id] = await create_klyqa_api_from_config(
+            hass, ConfigType(entry.data)
+        )
+        klyqa = hass.data[DOMAIN].entries[entry.entry_id]
+
+        if not klyqa or not await hass.async_add_executor_job(klyqa.login):
+            return
+
+    klyqa = hass.data[DOMAIN].entries[entry.entry_id]
+    if klyqa:
```

Comment:
Move API creating into the integration setup. This should not be done in the light platform.